src/rgw/rgw_op.cc

```diff
@@ -921,23 +921,9 @@ void rgw_build_iam_environment(rgw::sal::Driver* driver,
     s->env.emplace("aws:SecureTransport", "true");
   }
 
-  const auto remote_addr_param = s->cct->_conf->rgw_remote_addr_param;
-  if (remote_addr_param.length()) {
-    i = m.find(remote_addr_param);
-  } else {
-    i = m.find("REMOTE_ADDR");
-  }
-  if (i != m.end()) {
-    const string* ip = &(i->second);
-    string temp;
-    if (remote_addr_param == "HTTP_X_FORWARDED_FOR") {
-      const auto comma = ip->find(',');
-      if (comma != string::npos) {
-	temp.assign(*ip, 0, comma);
-	ip = &temp;
-      }
-    }
-    s->env.emplace("aws:SourceIp", *ip);
+  const auto ip = extract_remote_addr(s);
+  if (likely(ip.length())) {
+    s->env.emplace("aws:SourceIp", ip);
```

Comment:
when reviewing, found it useful to  log the selected IP address instead of doing gdb break 
```suggestion
    s->env.emplace("aws:SourceIp", ip);
    ldout(g_ceph_context, 21) << "Build IAM Environment: aws:SourceIp=" << ip << dendl;
```
@cbodely also made me wonder - would it be more correct to log this address in the `beast:` access log line?

Reply:
Probably we would then need to have this in `process_request` and then carry the IP for the logging and can be used only in iam evaluation - but I'm not sure if it would be sufficient as I guess because it's a header thingy it needs to be logged as a separate field and if it got to be empty - then an empty string is being logged always. probably this can be better placed in rgw ops logging which already logs the remote_addr based on this header.

Reply:
i assume that operators would expect the http access log line to contain the requesting client's ip, not that of a proxy

Reply:
Hi @cbodley - Could you please review the second commit (https://github.com/ceph/ceph/pull/55646/commits/5255dfcf29732703c200d54eee592ef0fd9f7093)?